Fantome.League/IO/WAD/WADFile.cs

```diff
@@ -58,9 +62,9 @@ public WADFile(Stream stream) : this()
                     throw new Exception("This is not a valid WAD file");
                 }
 
-                byte major = br.ReadByte();
-                byte minor = br.ReadByte();
-                if (major > 3)
+                _major = br.ReadByte();

```

Comment:
Can you please add `this.` before `_major` and `_minor` as those are members of this class and you are using `this.` in the Writing function.
This is just so that we have clean code.